crates/nu-parser/tests/test_parser.rs

```diff
@@ -784,6 +784,38 @@ pub fn parse_attributes_external_alias() {
     assert!(parse_error.contains("Encountered error during parse-time evaluation"));
 }
 
+#[test]
+pub fn parse_if_in_const_expression() {
+    // https://github.com/nushell/nushell/issues/15321
+    let engine_state = EngineState::new();
+    let mut working_set = StateWorkingSet::new(&engine_state);
+
+    working_set.add_decl(Box::new(Const));
```

Comment:
In general I would raise the same concern as in #15375 that those kinds of tests are pretty specific to the implementation and I personally would have been lazy and just used some thing like `nu!` or `run_test` to execute the code and check its output.